tool/fragment/src/main/java/org/openscience/cdk/fragment/FunctionalGroupsFinder.java

```diff
@@ -0,0 +1,1339 @@
+            this.type = aType;
+            this.bondIndex = anIndexInBond;
+            this.bondOrder = aConnectingBond.getOrder();
+            this.bondStereo = aConnectingBond.getStereo();
+            this.bondFlags = aConnectingBond.getFlags();
+        }
+        //
+        /**
+         * Returns the type, i.e. whether this carbon atom is aromatic or aliphatic.
+         *
+         * @return EnvironmentalCType enum constant
+         */
+        public EnvironmentalCType getType() {
+            return this.type;
+        }
+        //
+        /**
+         * Method for translating this instance back into a "real" IAtom instance when expanding the functional group
+         * environment, transferring all the cached properties, except the type(!).
+         *
+         * @param aTargetAtom marked functional group atom
+         * @param anEnvCAtom new carbon atom instance that should receive all the cached properties except the type(!);
+         *                   element, atom type "C" and implicit hydrogen count = 0 should be set already; type can later
+         *                   be set via .setIsAromatic(boolean);
+         * @return new bond connecting marked FG atom and environment atom in the correct order and with the cached properties
+         */
+        public IBond createBond(IAtom aTargetAtom, IAtom anEnvCAtom) {
+            IBond tmpBond = aTargetAtom.getBuilder().newInstance(IBond.class);
+            if (this.bondIndex == 0) {
+                tmpBond.setAtoms(new IAtom[] {anEnvCAtom, aTargetAtom});
+            }
+            else {
+                tmpBond.setAtoms(new IAtom[] {aTargetAtom, anEnvCAtom});
+            }
+            tmpBond.setOrder(this.bondOrder);
+            tmpBond.setStereo(this.bondStereo);
+            tmpBond.setFlags(this.bondFlags);
+            return tmpBond;
+        }
+    }
+    //
+    /**
+     * CDK logging tool instance for this class. Use FunctionalGroupsFinder.LOGGING_TOOL.setLevel(ILoggingTool.DEBUG);
+     * to activate debug messages.
+     */
+    public static final ILoggingTool LOGGING_TOOL = LoggingToolFactory.createLoggingTool(FunctionalGroupsFinder.class);
+    //
+    /**
+     * Property name for marking carbonyl carbon atoms via IAtom properties.
+     */
+    public static final String CARBONYL_C_MARKER = "FGF-Carbonyl-C";
+    //
+    /**
+     * Set of atomic numbers of nonmetal elements, namely hydrogen, carbon, nitrogen, oxygen, phosphorus, sulfur, selenium,
+     * halogens (fluorine, chlorine, bromine, iodine), and noble gases (helium, neon, argon, krypton, xenon, radon).
+     * Atoms of these elements are exclusively accepted in the input molecule if(!) the strict input restrictions are
+     * activated (turned off by default).
+     */
+    public static final Set<Integer> NONMETAL_ATOMIC_NUMBERS = new HashSet<>(Arrays.asList(1, 2, 6, 7, 8, 9, 10, 15, 16, 17, 18, 34, 35, 36, 53, 54, 86));
```

Comment:
IAtom.Hydrogen, IAtom.Helium, etc..

Reply:
Corrected in https://github.com/cdk/cdk/pull/1039/commits/8bde105371a29bb0a6fa439619652c41c2d2b5c0